sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/SearchAsyncClient.java

```diff
@@ -681,7 +678,7 @@ public String getEndpoint() {
                         try {
                             return new SimpleResponse<>(res, Utility.convertValue(res.getValue(), modelClass));
                         } catch (IOException ex) {
-                            throw logger.logExceptionAsError(
+                            throw LOGGER.logExceptionAsError(
```

Comment:
Unrelated to this PR - but we should avoid directly throwing RuntimeException if a more specific exception can be thrown. In this case, we could use `UncheckedIOException`.